src/main/webapp/map.js

```diff
@@ -135,4 +142,22 @@ function iconUrl(score){
     else url +="red";
 	url += "-dot.png";
 	return url;
+}
+
+function addWaterSystem(areaType, zipCode){
+    fetch('/water?area='+areaType+"&zip_code="+zipCode+"&town="+town+"&state="+state).then(response => response.json()).then((systems) => {
```

Comment:
It seems like your doGet handler in WaterPollutionServlet.java only uses the `town` and `state` parameters. Perhaps drop the other ones, or add handling for `area` and `zip_code` parameters back to doGet. This comment repeats the other comment I wrote on WaterPollutionServlet.java line 63.